stacks-signer/src/ping/processor.rs

```diff
@@ -0,0 +1,117 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        mpsc::Receiver,
+        Arc,
+    },
+    thread::{self, JoinHandle},
+    time::{Duration, Instant},
+};
+
+use super::Packet;
+
+/// the thread stores and computes requests.
+pub struct RTTProcessor {
+    /// Each entry is a unique Ping&Pongs request.
+    entries: HashMap<u64, EntryData>,
+    /// Get Pings from the user, get Pongs from the run_loop
+    rx: Receiver<(Instant, Packet)>,
+    keep_running: Arc<AtomicBool>,
+}
+
+///Holds the RTT distribution.
+struct EntryData {
+    tick: Instant,
+    average: RTTAverage,
+    variates: Vec<Duration>,
+}
+
+/// Calculate the average of an RTT ping-pong.
```

Comment:
The mean ping time isn't likely to be a meaningful statistical summary of the distribution of ping times, since there will likely be a long tail of response times.  Can you instead simply capture the RTTs and just dump them?  Let's leave statistical analysis to the caller.

Reply:
ack